peering.py

```diff
@@ -60,28 +60,84 @@ def withdraw(self, prefix, mux):
             pass
 
     def announce(self, prefix, spec):
-        for mux in spec['muxes']:
-            with open(self.__config_file(prefix, mux), 'w', encoding="utf8") as fd:
+        for mux in spec["muxes"]:
+            with open(self.__config_file(prefix, mux), "w", encoding="utf8") as fd:
                 fd.write(self.config_template.render(prefix=prefix, spec=spec))
 
     def reload_config(self):
         cmd = f"birdc -s {self.bird_sock}"
-        proc = subprocess.Popen(cmd.split(),
-                                stdin=subprocess.PIPE,
-                                stdout=subprocess.DEVNULL,
-                                stderr=subprocess.DEVNULL)
-        _stdout, _stderr = proc.communicate(b'configure\n')
+        proc = subprocess.Popen(
+            cmd.split(),
+            stdin=subprocess.PIPE,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        _stdout, _stderr = proc.communicate(b"configure\n")
 
 
-def main():
-    with open(sys.argv[1], 'r', encoding="utf8") as announcement_json_fd:
-        announcement = json.load(announcement_json_fd)
-    bird_cfg_dir = 'configs/bird'
-    bird_sock = 'var/bird.ctl'
-    schema_fn = 'configs/announcement_schema.json'
-    ctrl = AnnouncementController(bird_cfg_dir, bird_sock, schema_fn)
-    ctrl.deploy(announcement)
+class ExperimentController:
+    url = "http://127.0.0.1:7000"
 
+    def __init__(self, token, refresh=None, schema_fn=None):
```

Comment:
Please provide some documentation. Is the access token downloaded from the website's dashboard? What is `refresh`?

Reply:
It seems we can provide a default for `schema_fn`, as there is basically one file users will want to use as the schema.

Reply:
We should also pass a default to `schema_fn` in the `AnnouncementController` above.

Reply:
> Is the access token downloaded from the website's dashboard?
We can get it via browser:
![image](https://user-images.githubusercontent.com/22756244/192118153-57d31b57-418f-40d0-ba1a-b97b98f8b2be.png)
or via api request.

Reply:
I don't recognize that screenshot. Is it from the Website? The GET request seems to have failed, so not sure what it's intending to show. What I meant with my question is that users currently check their `AnnouncementCapabilities`, prefixes, deployed containers, and download their certs from the dashboard. So I thought we might make the token available right there... although we could discuss alternatives.

Reply:
This page is generated by django, the message above is just saying that the get request is not accepted, which is correct because to obtain the token you need to make a post request with username and password.

Reply:
With django-restframerok we also generate endpoints for users to send their experiments via web browser and obtain tokens from web browsers too (if they're already logged in ).